### DISCUSSION TITLE
media-preview-thumbnail overflowing

### DISCUSSION THREAD

**Discussion by jakubslambor** [muxinc/media-chrome#663] [Q&A] [answered]
Hi!

I have implemented the Preview Thumbnails in our player, however, it seems like the overflow prevention is not working for me 😢 can you help? In the youtube theme example, this is what does the trick:

![CleanShot 2023-06-05 at 15 26 33](https://github.com/muxinc/media-chrome/assets/63538910/75d8747b-8b12-4437-9488-23f0d6979403)

this does not generate for me:

![CleanShot 2023-06-05 at 15 28 00](https://github.com/muxinc/media-chrome/assets/63538910/35e0bf1e-7321-4f14-9016-c1317d98ecb0)

**Comment by luwes** [comment]
Hi, sorry for the later reply. Is this still an issue? thanks

**Reply by jakubslambor** [reply]
No worries. Unfortunately still a problem for me... I've tried stripping any custom styling/nesting and just doing the structure like

```vue
<media-controller>
  <mux-video>
    <track />
  </mux-video>

  <media-time-range class="h-8px w-full">
    <media-preview-thumbnail slot="preview" />
    <media-preview-time-display slot="preview" />
  </media-time-range>
<media-controller>
```

still, the thumbnail goes off screen - see screenshots above, the min-max code does not generate. Any ideas? Obviously not critical, but still I would like to have it working 😌 

<img width="607" alt="CleanShot 2023-07-21 at 15 32 22" src="https://github.com/muxinc/media-chrome/assets/63538910/8e7535a9-1029-4999-b994-6dfe73ea8897">

**Reply by luwes** [reply]
Would it be possible to provide a code sandbox with this overflow bug? that would help a lot in fixing this. thanks!

**Reply by jakubslambor** [reply]
Thank you for trying to help! I will try to create one, but it's a complex app, I hope I'll be able to extract the player to a sandbox.

**Reply by jakubslambor** [reply]
Sorry for the long delay, I got completely sidetracked. Unfortunately I cannot provide you with the sandbox as my codebase is too complex to rip just the player out 🥲 I've tried everything and the only way I could get it to work was if I put the player on the very first layer of the _entire_ app, like even outside of it (it's a Vue 3 project), right after body:

![CleanShot 2023-11-29 at 15 34 02@2x](https://github.com/muxinc/media-chrome/assets/63538910/3e0f8294-599a-4952-b309-d690b709aafb)

putting it any lower in the hierarchy breaks it for me:

![CleanShot 2023-11-29 at 15 32 40@2x](https://github.com/muxinc/media-chrome/assets/63538910/f8ce3b4e-d1ab-456d-a4d4-3297017a04d2)

From what I can see in the source code, the issue is that in the `#getBoxPosition` function in `media-time-range.js`, the `bounds` const is most likely wrongly set/undefined, therefore `boxMin` and `boxMax` are `NAN`, and so the returned position ends up being just the one defined on line [115](https://github.com/muxinc/media-chrome/blob/9a6e0d76f6caa5635d3af0f247411ad6a8e99065/src/js/media-time-range.js#L515), which skips the overflow prevention. Any ideas on what to wrap the player in to make this work? How is the mediaBoundsRect determined?

**Reply by jakubslambor** [reply, marked as answer]
This has been fixed at some point 🙌
